sync/sync.go

```diff
@@ -167,10 +166,11 @@ func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stat
 				s.log.Warnw("Sanity checks failed", "number", block.Number, "hash", block.Hash.ShortString())
 				resetStreams()
 				return
+			} else {
+				s.listener.OnSyncStepDone(OpVerify, block.Number, time.Since(verifyTimer))
```

Comment:
this should be moved back to it's original position, before the `return func()` statement.

Reply:
but  @joshklop commented in [this](https://github.com/NethermindEth/juno/pull/1387#pullrequestreview-1702853868
) to change s.listener.OnSyncStepDone(OpVerify

Reply:
I think he meant something different. If you move this line to here, `time.Since(verifyTimer)` won't yield the correct value.

Reply:
We still need to check `if err == nil` before calling `OnSyncStepDone(OpVerify`